src/arkode/arkode_impl.h

```diff
@@ -330,22 +330,25 @@ struct ARKodeMemRec
   void                       *step_mem;
 
   /* N_Vector storage */
-  N_Vector ewt;           /* error weight vector                             */
-  N_Vector rwt;           /* residual weight vector                          */
-  booleantype rwt_is_ewt; /* SUNTRUE if rwt is a pointer to ewt              */
-  N_Vector ycur;          /* pointer to user-provided solution memory; used
-                             as evolving solution by the timestepper modules */
-  N_Vector yn;            /* solution from the last successful step          */
-  N_Vector fn;            /* full IVP right-hand side from last step         */
-  N_Vector tempv1;        /* temporary storage vectors (for local use and by */
-  N_Vector tempv2;        /* time-stepping modules)                          */
+  N_Vector ewt;              /* error weight vector                           */
+  N_Vector rwt;              /* residual weight vector                        */
+  booleantype rwt_is_ewt;    /* SUNTRUE if rwt is a pointer to ewt            */
+  N_Vector ycur;             /* pointer to user-provided solution memory; used
+                                as evolving solution by the time stepper
+                                modules */
+  N_Vector yn;               /* solution from the last successful step        */
+  N_Vector fn;               /* full IVP right-hand side from last step       */
+  sunbooleantype fn_current; /* SUNTRUE if fn has been evaluated at yn        */
```

Comment:
Personally, I like to name boolean variables in a way that suggests that they are booleans. In this case, I would prefer `fn_is_current`.